src/Module/Item/Ignored.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace Friendica\Module\Item;
+
+use Friendica\App;
+use Friendica\BaseModule;
+use Friendica\Core\L10n\L10n;
+use Friendica\Core\Session;
+use Friendica\Database\Database;
+use Friendica\Model\Item;
+use Friendica\Network\HTTPException;
+
+/**
+ * Module for ignoring threads or user items
+ */
+class Ignored extends BaseModule
+{
+	public static function rawContent()
+	{
+		/** @var L10n $l10n */
+		$l10n = self::getClass(L10n::class);
+
+		if (!Session::isAuthenticated()) {
+			throw new HttpException\ForbiddenException($l10n->t('Access denied.'));
+		}
+
+		/** @var App\Arguments $args */
+		$args = self::getClass(App\Arguments::class);
+		/** @var Database $dba */
+		$dba = self::getClass(Database::class);
+
+		$message_id = intval($args->get(1));
+
+		if (empty($message_id) || !is_int($message_id)) {
+			throw new HTTPException\BadRequestException();
+		}
+
+		$thread = Item::selectFirstThreadForUser(local_user(), ['uid', 'ignored'], ['iid' => $message_id]);
+		if (!$dba->isResult($thread)) {
+			throw new HTTPException\BadRequestException();
+		}
+
+		// Numeric values are needed for the json output further below
+		$ignored = !empty($thread['ignored']) ? 0 : 1;
+
+		if (!empty($thread['uid']) && $thread['uid'] != 0) {
```

Comment:
This can be simplified to 
```suggestion
		if (!$thread['uid'])) {
```
Since the field will always be present, we don't have to check for not set index.

Reply:
BTW: Possibly we should check that if the thread uid is non zero, if it is equal to ```local_user()```, just for some security.

Reply:
Ha, I think that's the point I "really" mean  I can use GUIDs of other users and then the `uid` of the retrieved thread record is the uid of the other user. But since we just check the existence, it is set `ignored` anyway

Reply:
The GUID of a post is unique per post. But every entry in the item (and thread) table has got it's own id/iid entry. So saying " I can use GUIDs of other users threads" isn't correct.

Reply:
Hm .. OK I've to test it again .. I could have sworn that I could enable/disable the `ignored` mark of a thread record, which wasn't related to the user itself ..

Reply:
In this module, `Item::selectFirstThreadForUser(local_user(), ...)` will only retrieve the thread of the current user or the public thread (`uid = 0`) so you couldn’t hijack other people’s threads based on the item ID.

Reply:
I tested it again with the DB-fixture https://github.com/friendica/friendica/blob/develop/friendica_test_data.sql 

- login as `admin` with `uid = 1`
- call https://friendica.local/ignored/6 for thread with `GUID = 6`
- the returned `$thread['uid']` is now `3`, because the thread record with GUID 6 is bound to a the UID 3 ..

That's the item query behind it:
```sql
SELECT `item`.`uid`, `item`.`id` AS `internal-iid`, `item`.`network` AS `internal-network`, `item`.`icid` AS `internal-icid`, `item`.`iaid` AS `internal-iaid`, `user-item`.`ignored` AS `internal-user-ignored`, `thread`.`ignored` FROM `thread` STRAIGHT_JOIN `item` ON `item`.`id` = `thread`.`iid` STRAIGHT_JOIN `contact` ON `contact`.`id` = `thread`.`contact-id`
				AND NOT `contact`.`blocked`
				AND ((NOT `contact`.`readonly` AND NOT `contact`.`pending` AND (`contact`.`rel` IN (2, 3)))
				OR `contact`.`self` OR `item`.`gravity` != 0 OR `contact`.`uid` = 0)
				STRAIGHT_JOIN `contact` AS `author` ON `author`.`id` = `thread`.`author-id` AND NOT `author`.`blocked`
				STRAIGHT_JOIN `contact` AS `owner` ON `owner`.`id` = `thread`.`owner-id` AND NOT `owner`.`blocked`
				LEFT JOIN `user-item` ON `user-item`.`iid` = `thread`.`iid` AND `user-item`.`uid` = 1
				LEFT JOIN `user-contact` AS `user-author` ON `user-author`.`cid` = `thread`.`author-id` AND `user-author`.`uid` = 1
				LEFT JOIN `user-contact` AS `user-owner` ON `user-owner`.`cid` = `thread`.`owner-id` AND `user-owner`.`uid` = 1 WHERE (`thread`.`iid` = ?) AND `thread`.`visible` AND NOT `thread`.`deleted` AND NOT `thread`.`moderated`
			AND (`user-item`.`hidden` IS NULL OR NOT `user-item`.`hidden`)
			AND (`user-author`.`blocked` IS NULL OR NOT `user-author`.`blocked`)
			AND (`user-author`.`ignored` IS NULL OR NOT `user-author`.`ignored` OR `item`.`gravity` != 0)
			AND (`user-owner`.`blocked` IS NULL OR NOT `user-owner`.`blocked`)
			AND (`user-owner`.`ignored` IS NULL OR NOT `user-owner`.`ignored` OR `item`.`gravity` != 0)  LIMIT 1
```
Where `thread.iid` is `6`

Reply:
Ooooooh, that’s not great.

Reply:
That's why I meant that checking the returned ```uid``` against the ```local_user()``` is a good idea :-)

The user related item class calls aren't meant to add some security layer to it. So this behaviour is expected.

Reply:
So why was it like this before then? 🤔

Reply:
Because I haven't thought about this problem before.